# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsContentUtils.cpp
# Commit: d6c4fdcf6a69
# Full Hash: d6c4fdcf6a69ebaa1b133ef8c57da9eef389f89a
# Author: Tom Schuster <evilpies@gmail.com>
# Date: 2022-10-20 21:51:26
# Regressor Bug: 1778565
# File Overlap Count: 1
# Description:
#   Bug 1778565 - Editor: Handle nsIFile transferable as blob. r=nika,masayuki
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155753
# ==============================================================================

diff -r 88976c36da0a -r d6c4fdcf6a69 dom/base/nsContentUtils.cpp
--- a/dom/base/nsContentUtils.cpp	Thu Oct 20 06:59:31 2022 -0400
+++ b/dom/base/nsContentUtils.cpp	Thu Oct 20 11:11:59 2022 +0000
@@ -8104,28 +8104,6 @@
       // Otherwise, handle this as a file.
       nsCOMPtr<BlobImpl> blobImpl;
       if (nsCOMPtr<nsIFile> file = do_QueryInterface(data)) {
-        // FIXME(bug 1778565): Historically, attempting to send a Blob over IPC
-        // in response to a sync message would lead to a crash, however this
-        // isn't an issue anymore. Unfortunately, code in HTMLEditor depends on
-        // the old behaviour, so we need to preserve this oddity, and hope that
-        // the caller is HTMLEditor and can handle a nsIInputStream.
-        if (aInSyncMessage) {
-          nsAutoCString type;
-          if (IsFileImage(file, type)) {
-            IPCDataTransferItem* item =
-                aIPCDataTransfer->items().AppendElement();
-            item->flavor() = type;
-            nsCString data;
-            SlurpFileToString(file, data);
-            // FIXME: This can probably be simplified once bug 1783240 lands, as
-            // `nsCString` will be implicitly serialized in shmem when sent over
-            // IPDL directly.
-            item->data() =
-                IPCDataTransferInputStream(BigBuffer(AsBytes(Span(data))));
-          }
-          continue;
-        }
-
         if (aParent) {
           bool isDir = false;
           if (NS_SUCCEEDED(file->IsDirectory(&isDir)) && isDir) {